src/backend/aspen/api/utils/phylo.py

```diff
@@ -60,6 +92,81 @@ def _sample_filter(sample: Sample, can_see_pi_group_ids: Set[int], system_admin:
     return sample.submitting_group_id in can_see_pi_group_ids
 
 
+# set which countries will be given color labels in the nextstrain viewer
+# noqa: E711 is vital for the SQL query to compile correctly.
+async def _set_countries(db: AsyncSession, tree_json: dict, phylo_run: PhyloRun):
+    # information stored in tree_json["meta"]["colorings"], which is an
+    # array of objects. we grab the index of the one for "country"
+    country_defines_index = None
+    for index, category in enumerate(tree_json["meta"]["colorings"]):
+        if category["key"] == "country":
+            country_defines_index = index
+
+    # Next we grab the tree's country and the nearest 15 countries
+    tree_location = phylo_run.group.default_tree_location
+    countries = [tree_location.country]
+    origin_subq = (
+        sa.select(Location.country, Location.latitude, Location.longitude)  # type: ignore
+        .where(
+            and_(
+                Location.division == None,  # noqa: E711
+                Location.location == None,  # noqa: E711
+                Location.country == tree_location.country,
+            )
+        )
+        .subquery()
+        .lateral()
```

Comment:
This looks good for the most part -- Only comment is that I think the `lateral` isn't doing anything for us here, and it might (?) be clearer to write this as a query on the locations table for the origin, with a subquery for neighbors, instead of a join between two subqueries. But that's more of a style choice since I suspect our DB will compile down most of these variations to the same execution plan.